mivideo/mivideo_extract.py

```diff
@@ -297,31 +296,45 @@ def _makeCourseData(resultDictionaries: Sequence[Dict], categoryFilter: str) ->
         We want only those IDs that are integers, not hexadecimal or other strings.
 
         :param resultDictionaries: a sequence of KalturaMediaEntry objects converted to dictionaries
-        :param categoryFilter: the base Kaltura category, usually `Canvas_UMich`
         :return:
         """
         courseData: pd.DataFrame = pd.DataFrame.from_records(
             resultDictionaries, columns=('id', 'categories',)
-        ).rename(columns={'id': 'media_id', 'categories': 'course_id', })
+        ).rename(columns={'id': 'media_id', })
 
-        courseData = courseData.assign(
-            course_id=courseData['course_id'].str.split(',')).explode('course_id')
+        # split category CSVs to multiple new rows
+        courseData = (
+            courseData
+                .assign(categories=courseData['categories'].map(lambda c: c.split(',')))
+                .explode('categories')
+        )
 
-        courseData['in_context'] = [
-            c.endswith('>InContext') for c in courseData['course_id']]
+        # remove rows whose category isn't in "Canvas_UMich>site>channels>" hierarchy
+        courseData = courseData[courseData['categories']
+            .map(lambda c: c.startswith('Canvas_UMich>site>channels>'))]
 
-        courseData['course_id'] = courseData['course_id'].str.replace(
-            r'^' + categoryFilter + r'.*>([0-9]+)(>InContext)?$',
-            lambda m: m.groups()[0], regex=True
-        )
+        courseData['in_context'] = courseData['categories'] \
+            .map(lambda c: c.endswith('>InContext'))
+
+        # get course ID from '>'-delimited category hierarchy string
+        columnIndex = 3
+        courseData['course_id'] = \
+            courseData['categories'].map(lambda c: (c + '>' * columnIndex).split('>')[columnIndex])
+
+        # categories have been processed, drop unneeded column
+        courseData = courseData.drop('categories', axis=1)  # 1: columns
+
+        # find and drop invalid, non-decimal course IDs
+        # (e.g., "Shared Repository", "5430bafe907cca901a0f11646470dd64244ebd5f", etc.)
+        validCourseIdIndex = courseData['course_id'].map(str.isdecimal)
```

Comment:
Any reason you're not using `isnumeric` here? Aren't we handling integers?

Reply:
Yes!  I'm glad you asked.  🙂

`isdecimal()` matches only the characters '0' through '9'.  `isnumeric()` also matches things like super- and subscript digits ('⁰'–'⁹' and '₀'–'₉'; `isdigit()` matches these, too), fractions (e.g., '⅔', '⅖'), and maybe others.  We don't want those characters.

RTFM (no offense) on those functions.  This example summarizes it well: 

```python
def tests(s):
    print('s:', s)
    print('s.isdecimal (): ', s.isdecimal ())
    print('s.isnumeric (): ', s.isnumeric ())
    print('s.isdigit (): ', s.isdigit ())
    print()

tests(u'36246')  #decimal characters
tests(u'⁹')  #superscript digit
tests(u'½¼')  #fractional value
tests(u'③④⑤')  # circled digits
```

Reply:
Having a hard time imagining these weird values in output, but OK, what you have is fine.

Reply:
Yes, I have a hard time imagining anyone would use those characters in MiVideo media categories, either.  But the point is, they could.  I don't know how well Kaltura prevents those characters.  Should they prevent them?  Maybe not.  Whether or not they're there, we don't want them in this case.  That's why `isdecimal()` is the right thing to use here.